src/Umbraco.Web.UI.Client/src/common/services/clipboard.service.js

```diff
@@ -1,203 +1,276 @@
+            return null;
+        }
+        var dataJSON;
+        var dataString = localStorageService.get(STORAGE_KEY);
+        if (dataString != null) {
+            dataJSON = JSON.parse(dataString);
+        }
+        
+        if(dataJSON == null) {
+            dataJSON = new Object();
+        }
+        
+        if(dataJSON.entries === undefined) {
+            dataJSON.entries = [];
+        }
+        
+        return dataJSON;
+    }
+    
+    var saveStorage = function(storage) {
+        var storageString = JSON.stringify(storage);
+        
+        try {
+            var storageJSON = JSON.parse(storageString);
+            localStorageService.set(STORAGE_KEY, storageString);
+            
+            eventsService.emit("clipboardService.storageUpdate");
+            
+            return true;
+        } catch(e) {
+            return false;
+        }
+        
+        return false;
+    }
+
+    var prepareEntryForStorage = function(entryData) {
+
+        var shallowCloneData = Object.assign({}, entryData);// Notice only a shallow copy, since we dont need to deep copy. (that will happen when storing the data)
+        delete shallowCloneData.key;
+        delete shallowCloneData.$$hashKey;
+        
+        return shallowCloneData;
+    }
+    
+    
+    var service = {};
+    
+    /**
+    * @ngdoc method
+    * @name umbraco.services.clipboardService#copy
+    * @methodOf umbraco.services.clipboardService
+    *
+    * @param {string} type A string defining the type of data to storing, example: 'elementType', 'contentNode'
+    * @param {string} alias A string defining the alias of the data to store, example: 'product'
+    * @param {object} entry A object containing the properties to be saved, this could be the object of a ElementType, ContentNode, ...
+    * @param {string} displayLabel (optional) A string swetting the label to display when showing paste entries.
+    *
+    * @description
+    * Saves a single JS-object with a type and alias to the clipboard.
+    */
+    service.copy = function(type, alias, data, displayLabel) {
+        
+        var storage = retriveStorage();
+
+        var uniqueKey = data.key || data.$$hashKey || console.error("missing unique key for this content");
+        
+        // remove previous copies of this entry:
+        storage.entries = storage.entries.filter(
+            (entry) => {
+                return entry.unique !== uniqueKey;
+            }
+        );
+        
+        var entry = {unique:uniqueKey, type:type, alias:alias, data:prepareEntryForStorage(data), label:displayLabel || data.name, icon:iconHelper.convertFromLegacyIcon(data.icon)};
+        storage.entries.push(entry);
+        
+        if (saveStorage(storage) === true) {
+            notificationsService.success("Clipboard", "Copied to clipboard.");
+        } else {
+            notificationsService.success("Clipboard", "Couldnt copy this data to clipboard.");
```

Comment:
Is this a bit confusing/misleading UX to give a green/success message when an error happened @nielslyngsoe ?

Reply:
Yep, should be changed